src/util/prolog/prologResult.ts

```diff
@@ -1,6 +1,9 @@
 export class PrologResult {
-	private readonly results: string[][];
 	private readonly rawResults: string[];
+	private readonly results: string[][];
+
+	// First Group will be the variable value. Doesn't fetch the variable names
+	private readonly prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
```

Comment:
No need to store this in every instance of the class, since it's effectively a constant. You can put it into a not-exported constant outside the class:
```
const prologAnswerPattern = " = (\\w*|\\[.*\\])?(,| ;|.)";
```
You might even want to create a regex literal:
```
const prologAnswerPattern = / = (\w*|\[.*\])?(,| ;|.)/;
```
This saves a few backslashes (which can be donated to Microsoft, their OS uses a lot of them ^^).

Reply:
Edit: I see you need to add the variable name to it. No regex literal then.

Reply:
Exactly! I need to keep it as a string to enhance it later. But it sure makes sense to save it outside of the class.